contracts/FT-voting-simple/RIFGovernorFT.sol

```diff
@@ -13,15 +13,21 @@ contract RIFGovernorFT is
     GovernorVotes,
     GovernorVotesQuorumFraction
 {
-    ProposalTarget public immutable proposalTarget;
+    IProposalTarget public proposalTarget;
 
     constructor(IVotes _token)
         Governor('RIFGovernorFT')
         GovernorVotes(_token)
         GovernorVotesQuorumFraction(4)
-    {
-        // deploy proposal target smart contract
-        proposalTarget = new ProposalTarget();
+    {}
+
+    function updateProposalTarget(IProposalTarget _newTarget) public {
+        // can not be private since its call will be encoded in proposals
```

Comment:
:+1: thanks for this comment!

Reply:
:thinking: wondering if the *only* reason here is because we need `solc` to generate the function signature within the ABI output.

If that is the case, we could probably change this to private (more gas efficient), and then in the test code and the client DApp code, we could use a modified ABI. For example using [ethers.js' ABI fragments](https://docs.ethers.io/v5/single-page/#/v5/api/utils/abi/fragments/)

```js
new ethers.Contract(
  governor.address,
  [
    ...governor.interface.fragments,
    `function updateProposalTarget(IProposalTarget _newTarget) public`,
  ],
  signer,
);
```